bot/exts/core/leaderboard.py

```diff
@@ -0,0 +1,210 @@
+            content=":white_check_mark: Cleared all total game leaderboards.",
+            ephemeral=False
+        )
+
+        self.stop()
+
+    @ui.button(label="Cancel", style=ButtonStyle.grey, row=0)
+    async def cancel(self, _button: ui.Button, interaction: Interaction) -> None:
+        """Functionality ran when the clearing the total leaderboard has been denied."""
+        await interaction.response.send_message(
+            content=":x: Clearing cache aborted!",
+            ephemeral=False,
+        )
+        self.stop()
+
+
+class Leaderboard(commands.Cog):
+    """Cog for getting game leaderboards."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def get_rank(n: int) -> str:
+        """Get the rank in form of emoji (first 3 places) or ordinal number for `n`."""
+        first_3 = {1: ":first_place:", 2: ":second_place:", 3: ":third_place:"}
+        if n in first_3:
+            return first_3[n]
+
+        suffix = ["th", "st", "nd", "rd", "th"][min(n % 10, 4)]
+        if 11 <= (n % 100) <= 13:
+            suffix = "th"
+        return str(n) + suffix
+
+    async def make_leaderboard(
+        self,
+        author: discord.Member,
+        cached_leaderboard: list[RedisCache]
+    ) -> discord.Embed:
+        """Make a discord embed for the current top 10 members in the cached leaderboard."""
+        if len(cached_leaderboard) == 1:
+            unsorted_lb = await cached_leaderboard[0].to_dict()
+        else:
+            unsorted_lb = {}
+            for lb in cached_leaderboard:
+                async for member, points in lb.to_dict():
+                    exists = unsorted_lb.get(member)
+
+                    if exists:
+                        unsorted_lb[member] += points
+                    else:
+                        unsorted_lb[member] = points
+
+        sorted_lb = OrderedDict(sorted(unsorted_lb.items(), key=lambda kv: kv[1], reverse=True))
+        top_ten = itertools.islice(sorted_lb.items(), 0, 10)
+
+        try:
+            author_rank = list(sorted_lb).index(author.id) + 1
+        except ValueError:
+            author_rank = 0
+
+        lines = []
+        for index, (member_id, score) in enumerate(top_ten, start=1):
+            rank = self.get_rank(index)
+            buffer = 4 if index >= 4 else len(rank)+1
+            formatted_rank = format(rank, f"\N{BRAILLE PATTERN BLANK}<{buffer}")
+            formatted_score = format(score, "\N{BRAILLE PATTERN BLANK}<4")
+            lines.append(f"{formatted_rank}{formatted_score}— <@{member_id}>")
+
+        embed = discord.Embed(
+            title="Top 10",
+            description="\n".join(lines) if lines else "(no entries yet)",
+            colour=Colour.from_rgb(255, 230, 102),
+        )
+        embed.set_footer(
+            text=(
+                f"Page 1/{math.ceil(len(sorted_lb) / 10)}  |"
+                f"  Your rank: {author_rank}/{len(sorted_lb)}"
```

Comment:
Maybe we could just *not* display the user's rank if they aren't on the leaderboard?
```suggestion
                f"  |  Your rank: {author_rank}/{len(sorted_lb)}" if author_rank else ""
```

Reply:
If we don't display the user's rank on the leaderboard there is no way for them to know where they stand, I could create a separate command to see someone's rank but I think we can just include it in the same command.

Reply:
The suggestion is to specifically hide it if the user isn't even on the leaderboard.